homeassistant/components/advantage_air/climate.py

```diff
@@ -175,7 +163,12 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
 
     async def async_set_fan_mode(self, fan_mode: str) -> None:
         """Set the Fan Mode."""
-        await self.async_update_ac({"fan": HASS_FAN_MODES.get(fan_mode)})
+        mode = (
+            ADVANTAGE_AIR_MYFAN
+            if fan_mode == FAN_AUTO and self._ac.get(ADVANTAGE_AIR_AUTOFAN_ENABLED)
+            else fan_mode
```

Comment:
This still means that if the user sets `FAN_AUTO` it will still try to set it.
Maybe that's possible but given the names of the constants I would take it that it's only possible to use auto if `ADVANTAGE_AIR_AUTOFAN_ENABLED` is set?

Reply:
The hardware has **two** different automatic fan modes, if the "AUTOFAN" mode is enabled in the hardware's settings then we use that, and if not we use the regular auto fan mode. This matches how the UI on the hardware works.